test/contrib/sam/test_model.py

```diff
@@ -0,0 +1,111 @@
+        inpt = torch.rand(1, 3, 77, 128, device=device, dtype=dtype)
+        model = Sam.from_pretrained('vit_b', device=device)
+        inpt = _pad_rb(inpt, model.image_encoder.img_size)
+        keypoints = torch.randint(0, min(inpt.shape[-2:]), (batch_size, N, 2), device=device).to(dtype=dtype)
+        labels = torch.randint(0, 1, (batch_size, N), device=device).to(dtype=dtype)
+
+        out = model(inpt, [{'points': (keypoints, labels)}], multimask_output)
+
+        C = 3 if multimask_output else 1
+        assert len(out) == inpt.size(0)
+        assert out[0].logits.shape == (batch_size, C, 256, 256)
+
+    def test_exception(self):
+        model = Sam.build('vit_b')
+        with pytest.raises(TypeError) as errinfo:
+            inpt = torch.rand(3, 1, 2)
+            model(inpt, [], False)
+        assert 'shape must be [[\'B\', \'3\', \'H\', \'W\']]. Got torch.Size([3, 1, 2])' in str(errinfo)
+
+        with pytest.raises(Exception) as errinfo:
+            inpt = torch.rand(2, 3, 1, 2)
+            model(inpt, [{}], False)
+        assert 'The number of images (`B`) should match with the length of prompts!' in str(errinfo)
+
+    @pytest.mark.parametrize('model_type', ['vit_b', 'vit_l', 'vit_h'])
+    def test_build(self, device, model_type):
+        model = Sam.build(model_type)
+        model = model.to(device=device)
+
+        assert isinstance(model, Sam)
+        assert next(model.parameters()).device == device
+
+    @pytest.mark.parametrize(
+        'model_type,checkpoint', [('vit_b', 'https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth')]
+    )
+    def test_from_pretrained(self, device, model_type, checkpoint):
+        model = Sam.from_pretrained(model_type, checkpoint, device)
+
+        assert isinstance(model, Sam)
+        assert next(model.parameters()).device == device
+
+    @pytest.mark.skip(reason='Unsupport at moment -- the code is not tested for trainning and had `torch.no_grad`')
+    def test_gradcheck(self, device):
+        ...
+
+    @pytest.mark.skip(reason='Unecessary test')
+    def test_module(self):
+        ...
+
+    @pytest.mark.skip(reason='Needs to be reviewed')
+    def test_dynamo(self, device, dtype, torch_optimizer):
+        img = torch.rand(1, 3, 128, 75, device=device, dtype=dtype)
+
+        op = Sam.build('vit_b')
+        op = op.to(device=device, dtype=dtype)
+        op_optimized = torch_optimizer(op)
+
+        img = _pad_rb(img, op.image_encoder.img_size)
+
+        expected = op(img, [{}], False)
+        actual = op_optimized(img, [{}], False)
+
+        self.assert_close(expected[0].logits, actual[0].logits)
+        self.assert_close(expected[0].scores, actual[0].scores)
+
+    @pytest.mark.parametrize(
+        'model_type,checkpoint', [('vit_b', 'https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth')]
+    )
+    def test_prediction(self, device, dtype, model_type, checkpoint):
```

Comment:
this is failing, I pushed it to see if maybe we should download an asset image for it? @edgarriba

Reply:
i'm not a big fan of having the weights here (~350Mb), but if we would use/test it 1 time, this will be already cached

then downloading a test image, shouldn't slow more the tests suite when compared with the weights

maybe we should add markers to these tests, and for default not run it (and run it just on CI)